### PR TITLE
En_Ishi

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Ishi/z_en_ishi.c [hunk 1]
```diff
@@ -34,109 +54,703 @@ const ActorInit En_Ishi_InitVars = {
...
+    } else if (this->actor.bgCheckFlags & 8) {
+        sp2C.x -= 2.0f * this->actor.velocity.x;
+        sp2C.y += 2.0f * this->actor.velocity.y;
+        sp2C.z -= 2.0f * this->actor.velocity.z;
+    }
+    func_800BBFB0(globalCtx, &sp2C, 140.0f, 10, 180, 90, 1);
+}
+
+void func_8095DF90(EnIshi* this, GlobalContext* globalCtx) {
+    if (!(ENISHI_GET_1(&this->actor)) && !ENISHI_GET_100(&this->actor)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Ishi/z_en_ishi.c:295]:
```suggestion
    if (!ENISHI_GET_1(&this->actor) && !ENISHI_GET_100(&this->actor)) {
```

File: src/overlays/actors/ovl_En_Ishi/z_en_ishi.c [hunk 2]
```diff
@@ -34,109 +54,703 @@ const ActorInit En_Ishi_InitVars = {
...
+        this->unk_197 |= 2;
+    }
+
+    Actor_ProcessInitChain(&this->actor, sInitChain[sp34]);
+
+    if (globalCtx->csCtx.state != 0) {
+        this->actor.uncullZoneForward += 1000.0f;
+    }
+
+    if ((this->actor.shape.rot.y == 0) && ((this->unk_197 & 2) == 0)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Ishi/z_en_ishi.c:391]:
```suggestion
    if ((this->actor.shape.rot.y == 0) && !(this->unk_197 & 2)) {
```

File: src/overlays/actors/ovl_En_Ishi/z_en_ishi.c [hunk 3]
```diff
@@ -34,109 +54,703 @@ const ActorInit En_Ishi_InitVars = {
...
+
+void func_8095E5AC(EnIshi* this) {
+    this->actionFunc = func_8095E5C0;
+}
+
+void func_8095E5C0(EnIshi* this, GlobalContext* globalCtx) {
+    if (Object_IsLoaded(&globalCtx->objectCtx, this->unk_196)) {
+        this->actor.objBankIndex = this->unk_196;
+        this->actor.flags &= ~0x10;
+        if (!(ENISHI_GET_8(&this->actor))) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Ishi/z_en_ishi.c:445]:
```suggestion
        if (!ENISHI_GET_8(&this->actor)) {
```

File: src/overlays/actors/ovl_En_Ishi/z_en_ishi.c [hunk 4]
```diff
@@ -34,109 +54,703 @@ const ActorInit En_Ishi_InitVars = {
...
+            BgCheck_EntityRaycastFloor5(&globalCtx->colCtx, &this->actor.floorPoly, &sp2C, &this->actor, &sp30);
+    }
+}
+
+void func_8095EA70(EnIshi* this) {
+    f32 sp24;
+
+    this->actor.velocity.x = Math_SinS(this->actor.world.rot.y) * this->actor.speedXZ;
+    this->actor.velocity.z = Math_CosS(this->actor.world.rot.y) * this->actor.speedXZ;
+    if (!(ENISHI_GET_1(&this->actor))) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Ishi/z_en_ishi.c:554]:
```suggestion
    if (!ENISHI_GET_1(&this->actor)) {
```

File: src/overlays/actors/ovl_En_Ishi/z_en_ishi.c [hunk 5]
```diff
@@ -34,109 +54,703 @@ const ActorInit En_Ishi_InitVars = {
...
+
+    this->actionFunc(this, globalCtx);
+}
+
+void func_8095F210(EnIshi* this, GlobalContext* globalCtx) {
+    s32 pad;
+    s32 sp28;
+
+    if ((this->actor.projectedPos.z <= 1200.0f) ||
+        (((this->unk_197 & 1) != 0) && (this->actor.projectedPos.z < 1300.0f))) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Ishi/z_en_ishi.c:696]:
```suggestion
        ((this->unk_197 & 1) && (this->actor.projectedPos.z < 1300.0f))) {
```

**Review by hensldm — APPROVED** [review]
Small cosmetic suggestions.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**